src/components/SearchFilterList/SearchFilter.js

```diff
@@ -0,0 +1,119 @@
+import { useState, useMemo } from 'react'
+import { useResponsive } from 'hooks/useResponsive'
+import { Box, CheckBox, Heading, Text } from 'grommet'
+import { Button as sharedButton } from 'components/shared/Button'
+import { SearchBox } from 'components/shared/SearchBox'
+import { formatString } from 'helpers/formatString'
+import { scrollToId } from 'helpers/scrollToId'
+import styled, { css } from 'styled-components'
+
+const ToggleButton = styled(sharedButton)`
+  border-bottom: 1px solid transparent;
+  ${({ theme }) => css`
+    &:hover {
+      border-bottom: 1px solid ${theme.global.colors.brand};
+    }
+  `}
+`
+
+export const SearchFilter = ({ filterGroup, label }) => {
+  const { setResponsive } = useResponsive()
+  const options = useMemo(() => {
+    return Object.entries(filterGroup)
+  }, [filterGroup])
+  const [filteredResult, setfilteredResult] = useState(options)
+  const [open, setOpen] = useState(false)
+  const [userInput, setUserInput] = useState('')
+  const maxCount = 5
+  const filterLength = options.length
+
+  const filterOptions = (val) => {
+    setUserInput(val)
+
+    if (val.trim() !== '') {
+      setfilteredResult(() =>
+        options.filter((option) =>
+          formatString(option[0]).toLowerCase().startsWith(val.toLowerCase())
+        )
+      )
+    } else {
+      setfilteredResult(options)
+    }
+  }
+
+  const getOptionsToRender = () =>
+    open ? filteredResult : filteredResult.slice(0, maxCount)
+
+  return (
+    <>
+      <Heading
+        level={4}
+        margin={{ bottom: 'xsmall' }}
+        id={label.toLowerCase()}
```

Comment:
I think this would be more consistent with best practices if we pass the element via a reference and not an ID.

```suggestion
  const scrollRef = useRef()
  return (
    <>
      <Heading
        level={4}
        margin={{ bottom: 'xsmall' }}
        ref={scrollRef}
```

Reply:
Since this repository is currently active development, it would be great if the [latest version](https://github.com/AlexsLemonade/refinebio-web/blob/1339f00b5ae83f4e13bade1a5bd8f094723b9b88/src/components/SearchResults/SearchFilterList/SearchFilter.js) of this component were reviewed. Perhaps there must be a better approach in handling such scenarios. Let's discuss 💡